File: c#/tema_2/ConcertPro/Dao/Repository/OrderRepository/OrderDbRepository.cs
using System.Collections.Generic;
using System.Data;
using ConcertPro.Dao.Model;
using log4net;

namespace ConcertPro.Dao.Repository.OrderRepository
{
    public class OrderDbRepository : IOrderRepository
    {
        private static readonly ILog Log = LogManager.GetLogger("OrderDbRepository");
        private readonly IDictionary<string, string> _properties;

        public OrderDbRepository(IDictionary<string, string> properties)
        {
            Log.InfoFormat("Initializing OrderDbRepository with properties: {0}", properties);
            _properties = properties;
        }

        public Order FindOne(long id)
        {
            throw new System.NotImplementedException();
        }

        public IEnumerable<Order> FindAll()
        {
            throw new System.NotImplementedException();
        }

        public void Save(Order order)
        {
            Log.Info("Saving Order Task" + order);   
            IDbConnection connection = DbUtils.DbUtils.GetConnection(_properties);

            using (IDbCommand command = connection.CreateCommand())
            {
                command.CommandText = "insert into orders (buyerName, spectacleId, numberOfSeats) values " +
                                      "(@buyerName, @spectacleId, @numberOfSeats)";
                
                IDbDataParameter buyerNameParameter = command.CreateParameter();
                buyerNameParameter.ParameterName = "@buyerName";
                buyerNameParameter.Value = order.BuyerName;
                command.Parameters.Add(buyerNameParameter);
                
                IDbDataParameter spectacleIdParameter = command.CreateParameter();
                spectacleIdParameter.ParameterName = "@spectacleId";
                spectacleIdParameter.Value = order.Spectacle.Id;
                command.Parameters.Add(spectacleIdParameter);
                
                IDbDataParameter numberOfSeatsParameter = command.CreateParameter();
                numberOfSeatsParameter.ParameterName = "@numberOfSeats";
                numberOfSeatsParameter.Value = order.NumberOfSeats;
                command.Parameters.Add(numberOfSeatsParameter);

                int result = command.ExecuteNonQuery();
                if (result == 0)
                {
                    Log.Error("Error DB: No Order Added");
                }
                else
                {
                    Log.Info("Saved " + result + " instances");
                }
            }
        }

        public void Update(Order entity, long id)
        {
            throw new System.NotImplementedException();
        }

        public void Delete(long id)
        {
            throw new System.NotImplementedException();
        }
    }
}

File: java/tema_6/src/main/java/ro/mpp2024/concertpro/dao/repository/order_repository/OrderDbRepository.java
package ro.mpp2024.concertpro.dao.repository.order_repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp2024.concertpro.dao.model.Order;
import ro.mpp2024.concertpro.dao.validator.OrderValidator;
import ro.mpp2024.concertpro.dao.validator.Validator;
import ro.mpp2024.concertpro.utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.Properties;

public class OrderDbRepository implements IOrderRepository {
    private final JdbcUtils jdbcUtils;
    private final Validator<Order> orderValidator;
    private static final Logger logger = LogManager.getLogger();

    public OrderDbRepository(Properties properties, Validator<Order> orderValidator) {
        logger.info("Initializing OrderDbRepository with properties: {} ", properties);
        this.jdbcUtils = new JdbcUtils(properties);
        this.orderValidator = orderValidator;
    }

    @Override
    public Order findOne(Long orderId) {
        return null;
    }

    @Override
    public Iterable<Order> findAll() {
        return null;
    }

    @Override
    public void save(Order order) {
        logger.traceEntry("saving task {}", order);
        this.orderValidator.validate(order);

        Connection connection = this.jdbcUtils.getConnection();
        try(PreparedStatement preparedStatement = connection.prepareStatement("insert into orders (buyerName, " +
                "spectacleId, numberOfSeats) values (?, ?, ?)")) {
            preparedStatement.setString(1, order.getBuyerName());
            preparedStatement.setLong(2, order.getSpectacle().getId());
            preparedStatement.setLong(3, order.getNumberOfSeats());

            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }
        catch (SQLException exception) {
            logger.error(exception);
            System.err.println("Error DB " + exception);
        }

        logger.traceExit();
    }

    @Override
    public void update(Order order, Long orderId) {
    }

    @Override
    public void delete(Long orderId) {
    }
}
